File: mobile_app/fix_currency.dart
import 'dart:io';

void main() {
  final dir = Directory('lib/screens');
  if (!dir.existsSync()) {
    print('dir not found');
    return;
  }
  
  final regex = RegExp(r'String _formatCurrency\(num value\) \{[\s\S]*?\.format\(value\)\.trim\(\);\s*\}');
  
  final replacement = '''String _formatCurrency(dynamic value) {
    num parsedValue = 0;
    if (value is num) {
      parsedValue = value;
    } else if (value is String) {
      parsedValue = double.tryParse(value) ?? 0;
    }
    return intl.NumberFormat.currency(locale: 'ar_IQ', symbol: '', decimalDigits: 0).format(parsedValue).trim();
  }''';

  for (var file in dir.listSync(recursive: true)) {
    if (file is File && file.path.endsWith('.dart')) {
      var content = file.readAsStringSync();
      if (content.contains('String _formatCurrency(num value)')) {
        content = content.replaceAll(regex, replacement);
        file.writeAsStringSync(content);
        print('Updated \${file.path}');
      }
    }
  }
}

File: mobile_app/fix_currency.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

const regex = /String _formatCurrency\(num value\) \{[\s\S]*?trim\(\);\s*\}/g;
const replacement = `String _formatCurrency(dynamic value) {
    num parsedValue = 0;
    if (value != null) {
      if (value is num) {
        parsedValue = value;
      } else if (value is String) {
        parsedValue = double.tryParse(value) ?? 0;
      }
    }
    return intl.NumberFormat.currency(locale: 'ar_IQ', symbol: '', decimalDigits: 0).format(parsedValue).trim();
  }`;

walkDir('lib/screens', (filePath) => {
  if (filePath.endsWith('.dart')) {
    let content = fs.readFileSync(filePath, 'utf8');
    if (content.includes('String _formatCurrency(num value)')) {
      let newContent = content.replace(regex, replacement);
      if (newContent !== content) {
        fs.writeFileSync(filePath, newContent);
        console.log('Successfully replaced in ' + filePath);
      } else {
        console.log('Found but failed to replace in ' + filePath);
      }
    }
  }
});
